library/src/permutation/hiptensor_permutation.cpp

```diff
@@ -143,29 +153,124 @@ hiptensorStatus_t hiptensorPermutation(const hiptensorHandle_t*           handle
         return errorCode;
     }
 
-    if(descA->mType == HIP_R_16F)
+    // For now, enumerate all known permutation kernels.
+    auto& instances = hiptensor::PermutationSolutionInstances::instance();
+    auto  solnQ     = instances->allSolutions();
+
+    if(solnQ.solutionCount() == 0)
     {
-        return hiptensor::detail::permuteByCk(alpha,
-                                              static_cast<const _Float16*>(A),
-                                              descA,
-                                              modeA,
-                                              static_cast<_Float16*>(B),
-                                              descB,
-                                              modeB,
-                                              typeScalar,
-                                              stream);
+        // No kernels found!
+        auto errorCode = HIPTENSOR_STATUS_INTERNAL_ERROR;
+        snprintf(msg,
+                 sizeof(msg),
+                 "Internal Error : No Kernels Found (%s)",
+                 hiptensorGetErrorString(errorCode));
+        logger->logError("hiptensorPermutation", msg);
+        return errorCode;
     }
-    else if(descA->mType == HIP_R_32F)
+
+    // Extract the solutions to the candidates vector.
+    auto candidates = toPermutationSolutionVec(solnQ.solutions());
+
+    int  nDims              = descA->mLengths.size();
+    auto ADataType          = descA->mType;
+    auto BDataType          = descB->mType;
+    auto AOp                = descA->mUnaryOp;
+    auto BOp                = descB->mUnaryOp;
+
+    // Query permutation solutions for the correct permutation operation and type
+    auto solutionQ = hiptensor::PermutationSolutionRegistry::Query{candidates}
```

Comment:
Is there a higher level query that can accept all of the params queried?

Reply:
Yes added in the latest commit .